nilearn/datasets.py

```diff
@@ -1539,25 +1538,61 @@ def load_harvard_oxford(atlas_name,
     regions: nibabel.Nifti1Image
         regions definition, as a label image.
     """
-    if atlas_name not in ("cort-maxprob-thr0-1mm", "cort-maxprob-thr0-2mm",
-                          "cort-maxprob-thr25-1mm", "cort-maxprob-thr25-2mm",
-                          "cort-maxprob-thr50-1mm", "cort-maxprob-thr50-2mm",
-                          "sub-maxprob-thr0-1mm", "sub-maxprob-thr0-2mm",
-                          "sub-maxprob-thr25-1mm", "sub-maxprob-thr25-2mm",
-                          "sub-maxprob-thr50-1mm", "sub-maxprob-thr50-2mm",
-                          "cort-prob-1mm", "cort-prob-2mm",
-                          "sub-prob-1mm", "sub-prob-2mm"
-                          ):
+    atlas_items = ("cort-maxprob-thr0-1mm", "cort-maxprob-thr0-2mm",
+                   "cort-maxprob-thr25-1mm", "cort-maxprob-thr25-2mm",
+                   "cort-maxprob-thr50-1mm", "cort-maxprob-thr50-2mm",
+                   "sub-maxprob-thr0-1mm", "sub-maxprob-thr0-2mm",
+                   "sub-maxprob-thr25-1mm", "sub-maxprob-thr25-2mm",
+                   "sub-maxprob-thr50-1mm", "sub-maxprob-thr50-2mm",
+                   "cort-prob-1mm", "cort-prob-2mm",
+                   "sub-prob-1mm", "sub-prob-2mm")
+    if atlas_name not in atlas_items:
         raise ValueError("Invalid atlas name: {0}".format(atlas_name))
 
-    filename = os.path.join(dirname, "HarvardOxford-") + atlas_name + ".nii.gz"
+    if dirname == None:
+        # find atlas + meta-data from local FSL installation path
+        if 'FSL_DIR' in os.environ.keys():
+            local_dir = os.path.join(os.environ['FSL_DIR'], 'data', 'atlases',
+                                     'HarvardOxford')
+            xls1_found = os.path.exists(os.path.join(local_dir, '..',
+                                        'HarvardOxford-Cortical.xml'))
+            xls2_found = os.path.exists(os.path.join(local_dir, '..',
+                                        'HarvardOxford-SubCortical.xml'))
+            if xls1_found and xls2_found:
+                dirname = local_dir
+
+    if dirname is not None:
```

Comment:
FYI information, all these cases are already handled in _get_dataset_dir, you shouldn't do that manually. I will fix that, take a look when you have time.